app/models/user_preference.rb

```diff
@@ -0,0 +1,6 @@
+class UserPreference < ActiveRecord::Base
+  validates_presence_of :preference_id, null:false
```

Comment:
Space missing after colon.<br>Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.